src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -5969,540 +5822,652 @@ int SketchObject::removeAxesAlignment(const std::vector<int>& geoIdList)
-             ++it) {
-            if ((*it)->Type == Sketcher::InternalAlignment && (*it)->Second == GeoId) {
-                switch ((*it)->AlignmentType) {
-                    case Sketcher::BSplineControlPoint:
-                        controlpoints[(*it)->InternalAlignmentIndex] = true;
-                        controlpointgeoids[(*it)->InternalAlignmentIndex] = (*it)->First;
-                        break;
-                    case Sketcher::BSplineKnotPoint:
-                        knotpoints[(*it)->InternalAlignmentIndex] = true;
-                        knotgeoids[(*it)->InternalAlignmentIndex] = (*it)->First;
-                        break;
-                    default:
-                        return -1;
-                }
-            }
+    for (auto& con : icon) {
+        delete con;
+    }
+
+    return incrgeo;// number of added elements
+}
+
+template <>
+int SketchObject::exposeInternalGeometryForType<Part::GeomBSplineCurve>(const int GeoId)
+{
+    const Part::Geometry* geo = getGeometry(GeoId);
+
+    const auto* bsp = static_cast<const Part::GeomBSplineCurve*>(geo);
+    // First we search what has to be restored
+    std::vector<int> controlpointgeoids(bsp->countPoles(), GeoEnum::GeoUndef);
+
+    std::vector<int> knotgeoids(bsp->countKnots(), GeoEnum::GeoUndef);
+
+    bool isfirstweightconstrained = false;
+
+    const std::vector<Sketcher::Constraint*>& vals = Constraints.getValues();
+
+    // search for existing poles
+    for (auto const& constr : vals) {
+        if (constr->Type != Sketcher::InternalAlignment || constr->Second != GeoId) {
+            continue;
         }
 
-        if (controlpoints[0]) {
-            // search for first pole weight constraint
-            for (std::vector<Sketcher::Constraint*>::const_iterator it = vals.begin();
-                 it != vals.end();
-                 ++it) {
-                if ((*it)->Type == Sketcher::Weight && (*it)->First == controlpointgeoids[0]) {
-                    isfirstweightconstrained = true;
-                }
-            }
+        switch (constr->AlignmentType) {
+        case Sketcher::BSplineControlPoint:
+            controlpointgeoids[constr->InternalAlignmentIndex] = constr->First;
+            break;
+        case Sketcher::BSplineKnotPoint:
+            knotgeoids[constr->InternalAlignmentIndex] = constr->First;
+            break;
+        default:
+            return -1;
         }
+    }
 
-        int currentgeoid = getHighestCurveIndex();
-        int incrgeo = 0;
+    if (controlpointgeoids[0] != GeoEnum::GeoUndef) {
+        // search for first pole weight constraint
+        auto it = std::find_if(vals.begin(),
+                               vals.end(),
+                               [&controlpointgeoids](const auto& constr) {return (constr->Type == Sketcher::Weight && constr->First == controlpointgeoids[0]);});
+        isfirstweightconstrained = (it != vals.end());

```

Comment:
https://en.cppreference.com/w/cpp/algorithm/all_any_none_of - couldn't `std::any_of` be used here?

Reply:
If `it` is not used anywhere else down the line, of course. This is some pretty old change when this would have been more changes, and I hadn't added tests at the time.